server/src/modules/maps/map-library.service.ts

```diff
@@ -0,0 +1,40 @@
+import {
+    BadRequestException,
+    Injectable,
+    NotFoundException
+} from '@nestjs/common';
+import { PrismaService } from '../repo/prisma.service';
+import {
+    Prisma
+} from '@prisma/client';
+
+
+@Injectable()
+export class MapLibraryService {
+    constructor( private prisma: PrismaService ) {}
+
+    async isMapInLibrary(userID: number, mapID: number) {
+        const mapWhere: Prisma.MapWhereInput = {
+            id: mapID
+        };
```

Comment:
Declaring this separately feels redundant, just for the findFirst just do `where: { id: mapID }`